src/HotChocolate/Data/src/Data/Projections/Optimizers/IsProjectedProjectionOptimizer.cs

```diff
@@ -17,55 +20,192 @@ public class IsProjectedProjectionOptimizer : IProjectionOptimizer
         SelectionSetOptimizerContext context,
         Selection selection)
     {
-        if (!(context.Type is ObjectType type &&
-                type.ContextData.TryGetValue(AlwaysProjectedFieldsKey, out var fieldsObj) &&
-                fieldsObj is string[] fields))
+        var type = (ObjectType) context.Type;
+        var alwaysProjectedFieldNames = (string[]) type.ContextData[AlwaysProjectedFieldsKey]!;
+
+        int aliasedFieldIndex = 0;
+
+        // TODO: cache these strings, because why not.
+        string GetCurrentAlias() => "__projection_alias_" + aliasedFieldIndex;
+
+        for (; aliasedFieldIndex < alwaysProjectedFieldNames.Length; aliasedFieldIndex++)
         {
-            return selection;
+            var alias = GetCurrentAlias();
+            var fieldName = alwaysProjectedFieldNames[aliasedFieldIndex];
+
+            if (context.IsFieldAlreadyInSelection(fieldName, alias))
```

Comment:
I don't really understand the purpose of that alias, I figured you did that for some edge case, so I just left the logic you wrote. I just moved the check into that method, because I wanted to reuse it.

Reply:
it includes fields in the projection